Stats tests fail in UTC+ timezones: date.today() vs UTC mismatch

## Bug

6 stats tests fail consistently in UTC+ timezones (e.g. UTC+2) when run after midnight local time but before midnight UTC:

```
FAILED tests/test_router_stats.py::test_streak_today_only
FAILED tests/test_router_stats.py::test_streak_consecutive_days
FAILED tests/test_router_stats.py::test_streak_zero_when_last_activity_was_two_days_ago
FAILED tests/test_router_stats.py::test_longest_streak
FAILED tests/test_router_stats.py::test_activity_includes_vocabulary_events
FAILED tests/test_router_stats.py::test_progress_update_multiple_chapters_all_logged
```

## Root cause

The tests and the production code use different date bases:

| Code location | Date source |
|---|---|
| `_insert_history_at` helper | `date.today()` — machine local time |
| `test_activity_includes_vocabulary_events` | `date.today().isoformat()` — local |
| `test_progress_update_multiple_chapters_all_logged` | `date.today().isoformat()` — local |
| `reading_history.read_at` DEFAULT | `CURRENT_TIMESTAMP` — UTC (SQLite) |
| Activity SQL `WHERE ts >= DATE('now', '-365 days')` | `DATE('now')` — UTC (SQLite) |
| Streak calculation in `get_user_stats` | `datetime.now(timezone.utc).date()` — UTC |

When the machine is in UTC+2 and it's 23:30 local (01:30 next day UTC), `date.today()` returns tomorrow's UTC date. The streak calculation checks yesterday's UTC date but finds tomorrow's — streak reports 0 instead of 1.

## Fix

Change `_insert_history_at` and the two activity-date checks to use `datetime.now(timezone.utc).date()` instead of `date.today()`, aligning with the UTC-based production code.

--- a/backend/tests/test_router_stats.py
+++ b/backend/tests/test_router_stats.py
@@ -91,8 +91,9 @@ async def test_stats_totals_reflect_user_data(client, test_user, tmp_db):
 # ── Streak calculation ────────────────────────────────────────────────────────
 
 async def _insert_history_at(db_path, user_id, book_id, day_offset):
-    """Insert a reading_history row with a synthetic timestamp."""
-    target_date = date.today() - timedelta(days=day_offset)
+    """Insert a reading_history row with a synthetic timestamp (UTC-based)."""
+    from datetime import timezone, datetime as _dt
+    target_date = _dt.now(timezone.utc).date() - timedelta(days=day_offset)
     ts = f"{target_date.isoformat()} 12:00:00"
     async with aiosqlite.connect(db_path) as db:
         await db.execute(
@@ -154,7 +155,8 @@ async def test_activity_includes_vocabulary_events(client, test_user, tmp_db):
     await save_word(test_user["id"], "loquacious", BOOK_ID, 0, "A loquacious narrator.")
     resp = await client.get("/api/user/stats")
     activity = resp.json()["activity"]
-    today = date.today().isoformat()
+    from datetime import timezone, datetime as _dt
+    today = _dt.now(timezone.utc).date().isoformat()
     counts = {a["date"]: a["count"] for a in activity}
     assert today in counts
     assert counts[today] >= 1
@@ -206,7 +208,8 @@ async def test_progress_update_multiple_chapters_all_logged(client, test_user, t
 
     # 4 reading events → activity count for today >= 4
     resp = await client.get("/api/user/stats")
-    today = date.today().isoformat()
+    from datetime import timezone, datetime as _dt
+    today = _dt.now(timezone.utc).date().isoformat()
     activity = {a["date"]: a["count"] for a in resp.json()["activity"]}
     assert activity.get(today, 0) >= 4
 

